components/mediation-monitor/mediation-data-publisher/org.wso2.carbon.prometheus.publisher/src/main/java/org/wso2/carbon/prometheus/publisher/model/PrometheusMetric.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.prometheus.publisher.model;
+
+import org.wso2.carbon.prometheus.publisher.util.PrometheusPublisherConstants;
+
+public class PrometheusMetric {
+
+    private static final String DELIMITER = " ";
+    private static final String NEW_LINE = "\n";
+    private String metricName;
+    private String help;
+    private String type;
+    private Double metricValue;
+    private String propertyString;
+
+    /**
+     * Formats the metric data into Prometheus format
+     *
+     * @param metric Metric object
+     * @return Formatted metric string
+     */
+    public static String getFormattedMetric(PrometheusMetric metric) {
+
+        if (metricDataExist(metric)) {
+            return formatMetricHelp(metric) +
+                    formatMetricType(metric) +
+                    metric.getMetricName() +
+                    metric.getPropertyString() +
+                    DELIMITER +
+                    metric.getMetricValue()
+                    + NEW_LINE;
+        }
+        return "";
+    }
+
+    private static boolean metricDataExist(PrometheusMetric metric) {
+
+        return valuesExist(metric.getMetricName()) && metric.getMetricValue() != null;
```

Comment:
We have the null check in valuesExist method as well. Therefore do we need to have it again?

Reply:
The valueExists method validates the name of the metric, while the null check validates the value of the metric.

Reply:
:+1: I missed that